sdk/language-support/ts/codegen/tests/ts/build-and-lint-test/src/__tests__/test.ts

```diff
@@ -237,369 +237,416 @@ describe("decoders for recursive types do not loop", () => {
-    enum: buildAndLint.Main.Color.Red,
-    enumList: [
-      buildAndLint.Main.Color.Red,
-      buildAndLint.Main.Color.Blue,
-      buildAndLint.Main.Color.Yellow,
-    ],
-    enumList2: ["Red", "Blue", "Yellow"],
-    optcol1: { tag: "Transparent1", value: {} },
-    optcol2: { tag: "Color2", value: { color2: "Red" } }, // 'Red' is of type Color
-    optcol3: { tag: "Color2", value: { color2: buildAndLint.Main.Color.Blue } }, // Color.Blue is of type 'Color'
-    variant: {
-      tag: "Add",
-      value: { _1: { tag: "Lit", value: "1" }, _2: { tag: "Lit", value: "2" } },
-    },
-    optionalVariant: {
-      tag: "Add",
-      value: { _1: { tag: "Lit", value: "1" }, _2: { tag: "Lit", value: "2" } },
-    },
-    sumProd: { tag: "Corge", value: { x: "1", y: "Garlpy" } },
-    optionalSumProd: { tag: "Corge", value: { x: "1", y: "Garlpy" } },
-    parametericSumProd: {
-      tag: "Add2",
-      value: {
-        lhs: { tag: "Lit2", value: "1" },
-        rhs: { tag: "Lit2", value: "2" },
-      },
-    },
-    optionalOptionalParametericSumProd: [
-      {
+      sumProd: { tag: "Corge", value: { x: "1", y: "Garlpy" } },
+      optionalSumProd: { tag: "Corge", value: { x: "1", y: "Garlpy" } },
+      parametericSumProd: {
         tag: "Add2",
         value: {
           lhs: { tag: "Lit2", value: "1" },
           rhs: { tag: "Lit2", value: "2" },
         },
       },
-    ],
-    n0: "3.0", // Numeric 0
-    n5: "3.14159", // Numeric 5
-    n10: "3.1415926536", // Numeric 10
-    rec: { recOptional: null, recList: [], recTextMap: {} },
-    voidRecord: null,
-    voidEnum: null,
-    genMap: map,
-  };
-  const allTypesContract = await aliceLedger.create(
-    buildAndLint.Main.AllTypes,
-    allTypes,
-  );
-  expect(allTypesContract.payload).toEqual(allTypes);
-  expect(allTypesContract.key).toBeUndefined();
+      optionalOptionalParametericSumProd: [
+        {
+          tag: "Add2",
+          value: {
+            lhs: { tag: "Lit2", value: "1" },
+            rhs: { tag: "Lit2", value: "2" },
+          },
+        },
+      ],
+      n0: "3.0", // Numeric 0
+      n5: "3.14159", // Numeric 5
+      n10: "3.1415926536", // Numeric 10
+      rec: { recOptional: null, recList: [], recTextMap: {} },
+      voidRecord: null,
+      voidEnum: null,
+      genMap: map,
+    };
+    const allTypesContract = await aliceLedger.create(AllTypes, allTypes);
+    expect(allTypesContract.payload).toEqual(allTypes);
+    expect(allTypesContract.key).toBeUndefined();
+
+    const allTypesContracts = await aliceLedger.query(AllTypes);
+    expect(allTypesContracts).toEqual([allTypesContract]);
 
-  const allTypesContracts = await aliceLedger.query(buildAndLint.Main.AllTypes);
-  expect(allTypesContracts).toEqual([allTypesContract]);
+    const allTypesArchived = await aliceLedger.archive(
+      AllTypes,
+      allTypesContract.contractId,
+    );
+    expect(allTypesArchived.contractId).toEqual(allTypesContract.contractId);
```

Comment:
Clean up contracts, as the tests share a ledger and they'd otherwise interfere with one another's results.